lib/src/kbasesearchengine/main/NarrativeInfoDecorator.java

```diff
@@ -81,12 +86,17 @@ public SearchTypesOutput searchTypes(final SearchTypesInput params, final String
     public SearchObjectsOutput searchObjects(final SearchObjectsInput params, final String user)
             throws Exception {
         SearchObjectsOutput searchObjsOutput = searchInterface.searchObjects(params, user);
+
+        if(searchObjsOutput.getAdditionalProperties().get(removedGuids) == null && System.getProperty(removedGuidsEnv) != null){
```

Comment:
I would require an exact value rather than just non-null for the environment variable to avoid unexpected use. `true` would work. Also namespace the env var a bit like `KBASE_SEARCH_SHOW_REMOVED_GUIDS`.